Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```diff
@@ -0,0 +1,414 @@
+        
+        /// The utility network for this sample.
+        private var utilityNetwork = UtilityNetwork(url: .featureServiceURL)
+        
+        /// The service geodatabase used to create the feature layer.
+        private var serviceGeodatabase: ServiceGeodatabase!
+        
+        /// The service geodatabase feature layers.
+        private var layers: [FeatureLayer] = []
+        
+        /// The last element that was added to the list of filter barriers.
+        ///
+        /// When an element contains more than one terminal, the user should be presented with the
+        /// option to select a terminal. Keeping a reference to the last added element provides ease
+        /// of access to save the user's choice.
+        @Published var lastAddedElement: UtilityElement?
+        
+        /// The current tracing related activity.
+        @Published var tracingActivity: TracingActivity = .loadingServiceGeodatabase
+        
+        /// The base trace parameters.
+        @Published var traceParameters = UtilityTraceParameters(traceType: .isolation, startingLocations: [])
+        
+        /// The point geometry of the starting location.
+        @Published var startingLocationPoint: Point!
+        
+        /// The filter barrier categories.
+        @Published var filterBarrierCategories: [UtilityCategory] = []
+        
+        /// The selected filter barrier category.
+        @Published var selectedCategory: UtilityCategory?
+        
+        /// The previously selected filter barrier category.
+        @Published var previousCategory: UtilityCategory?
+        
+        /// A Boolean value indicating whether to include isolated features in the
+        /// trace results when used in conjunction with an isolation trace.
+        @Published var includesIsolatedFeatures = true
+        
+        /// A Boolean value indicating whether the trace is completed.
+        @Published var traceCompleted = false
+        
+        /// A Boolean value indicating whether the filter barrier categories have loaded.
+        @Published var categoriesLoaded = false
+        
+        /// A Boolean value indicating if tracing is enabled.
+        @Published var traceEnabled = false
+        
+        /// A Boolean value indicating if the reseting the trace is enabled.
+        @Published var resetEnabled = false
+        
+        /// A Boolean value indicating if the user has added filter barriers to the trace parameters.
+        @Published var hasFilterBarriers = false
+        
+        /// A Boolean value indicating whether the terminal selection menu is open.
+        @Published var terminalSelectorIsOpen = false
+        
+        /// A Boolean value indicating whether the configuration sheet is disabled.
+        @Published var configurationSheetDisabled = false
+        
+        /// The status text to display to the user.
+        @Published var statusText: String = "Loading Utility Network…"
```

Comment:
Can the model set the status text if the property is on the view? The text is related to the view, but the content is based on what is happening in the model.

Reply:
> the content is based on what is happening in the model

Correct. The view should observe the state of the model and update the UI (including any text displayed) appropriately. So the model should update its own state in a way that the view can observe and respond to (which I think it is) and then the view should be the one observing those state changes and updating the status text based on the current state of the model.

Reply:
I'm not sure how to keep the status text on the view since the status text uses string interpolation for properties on the model including the selected category name and selected terminal name. An enum on the view could be used to manage the status text like [this](https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/v.next/Shared/Samples/Create%20buffers%20around%20points/CreateBuffersAroundPointsView.swift#L265-L281), but since an enum can't contain a stored property the selected category name and terminal name can't be included. A few other samples also have the status text on the model and update it there.

Reply:
I also have confusion on this topic, and there are a few different situations.

1. The status text doesn't depend on any data from the model.

In this case, we can use sth like `enum Status` and update the `Text` views based on its case.

2. The status text depends on the data from the model.

In this case, we can still set an associate value (presumably a `String`) to the case. In the view however, while the other part of the status text can be localized, the associated value cannot be. Which I think is acceptable, given that the value reflects the status of the data, and doesn't effectively need to be localized.

- The status text idea in general

From there it might be a slippery slope to think that "well, if the status text is just an internal state, it is OK to be not-localizable". After all, it is a value, just happens to be a `String`. It can also be a `Int` status code, etc. I have to confess that I took a few shortcuts in the previous samples with this thinking in mind. In hindsight, I would rather have not done that, but keep the clear data model separation instead.

- Moving forward

    - We can design more intuitive UI to avoid status text overlay
    - Try to make the model statuses independent of its properties
    - If it has to use a property, try to pass the original type in the associated value (for instance, Geometry.Type vs String)
    - If there is no easy way, passing an associated `String` is a last resort